File: serverless-aws/python/handled_exception/handled_exception.py
# /**
# *******************************************************************************
# * File name   : handled_exception.py
# * Description : This file contains code that instruments handled exception
# *******************************************************************************
# **/

# Import Sentry library
import sentry_sdk
from sentry_sdk.integrations.aws_lambda import AwsLambdaIntegration

# Configure Sentry SDK
sentry_sdk.init(
    dsn="<your DSN>",
    integrations=[AwsLambdaIntegration()]
)


def lambda_handler(event, context):
    """Lambda function which raises an exception

    Args:
        event (dict): Parameter to pass in event data to the handler.
        context (bootstrap.LambdaContext): Parameter to provide runtime information to the handler.
    """
    try:
        division_by_zero = 1 / 0
    except Exception as e:
        # handle ZeroDivisionError exception
        print(e)
        raise

File: aws-lambda/node/handled_exception/handled_exception.js
/**
 *******************************************************************************
 * File name   : handled_exception.js
 * Description : This file contains code that instruments handled exception.
 *
 *******************************************************************************
 **/

// Import the Sentry module.
const Sentry = require("@sentry/node");
// Configure the Sentry SDK.
Sentry.init({
  dsn: "<your DSN>",
});

// below is the faulty code, undefinedFun() function is not exist.

exports.handler = function (event, context, callback) {
  try {
    undefinedFunCall(); // call undefined function.
  } catch (e) {
    Sentry.captureException(e); // Capture the exception in the Sentry dashboard.
    Sentry.flush(2000);
  }

  const response = {
    statusCode: 200,
    body: "Handled exception",
  };

  callback(null, response);
};
